From a9dc95765b0e18855a25afe1dc8aa5820e0a8aa7 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Fri, 3 Oct 2025 17:18:12 +0200
Subject: [PATCH] ARM: dts: stm32: add phase tags marking on stm32mp15

bootph-all/bootph-some-ram/bootph-pre-ram as phase tags were added to
dt-schema (dtschema/schemas/bootph.yaml) to describe various node usage
during boot phases with DT.

Describe them for STM32MP15 boot devices.

Change-Id: Id7205fc09355f8cbcfdee032ed7741884c4dd638
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/496296
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm/boot/dts/st/stm32mp15-scmi.dtsi |  2 ++
 arch/arm/boot/dts/st/stm32mp151.dtsi     | 31 ++++++++++++++++++++++++
 2 files changed, 33 insertions(+)

diff --git a/arch/arm/boot/dts/st/stm32mp15-scmi.dtsi b/arch/arm/boot/dts/st/stm32mp15-scmi.dtsi
index f113ba003673..a7a3e6634c22 100644
--- a/arch/arm/boot/dts/st/stm32mp15-scmi.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp15-scmi.dtsi
@@ -7,6 +7,7 @@
 / {
 	firmware {
 		optee: optee {
+			bootph-all;
 			compatible = "linaro,optee-tz";
 			method = "smc";
 			interrupt-parent = <&intc>;
@@ -14,6 +15,7 @@ optee: optee {
 		};
 
 		scmi: scmi {
+			bootph-some-ram;
 			compatible = "linaro,scmi-optee";
 			#address-cells = <1>;
 			#size-cells = <0>;
diff --git a/arch/arm/boot/dts/st/stm32mp151.dtsi b/arch/arm/boot/dts/st/stm32mp151.dtsi
index b7f870f1625d..a3a78ced375b 100644
--- a/arch/arm/boot/dts/st/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp151.dtsi
@@ -29,16 +29,19 @@ cpu0: cpu@0 {
 	};
 
 	cpu0_opp_table: cpu0-opp-table {
+		bootph-pre-ram;
 		compatible = "operating-points-v2";
 		opp-shared;
 
 		opp-650000000 {
+			bootph-pre-ram;
 			opp-hz = /bits/ 64 <650000000>;
 			opp-microvolt = <1200000>;
 			opp-supported-hw = <0x3>;
 		};
 
 		opp-800000000 {
+			bootph-pre-ram;
 			opp-hz = /bits/ 64 <800000000>;
 			opp-microvolt = <1350000>;
 			opp-supported-hw = <0x2>;
@@ -59,6 +62,7 @@ arm_wdt: watchdog {
 	};
 
 	psci {
+		bootph-some-ram;
 		compatible = "arm,psci-1.0";
 		method = "smc";
 	};
@@ -82,31 +86,37 @@ timer {
 	};
 
 	clocks {
+		bootph-all;
 		clk_csi: clk-csi {
+			bootph-all;
 			#clock-cells = <0>;
 			compatible = "fixed-clock";
 			clock-frequency = <4000000>;
 		};
 
 		clk_hse: clk-hse {
+			bootph-all;
 			#clock-cells = <0>;
 			compatible = "fixed-clock";
 			clock-frequency = <24000000>;
 		};
 
 		clk_hsi: clk-hsi {
+			bootph-all;
 			#clock-cells = <0>;
 			compatible = "fixed-clock";
 			clock-frequency = <64000000>;
 		};
 
 		clk_lse: clk-lse {
+			bootph-all;
 			#clock-cells = <0>;
 			compatible = "fixed-clock";
 			clock-frequency = <32768>;
 		};
 
 		clk_lsi: clk-lsi {
+			bootph-all;
 			#clock-cells = <0>;
 			compatible = "fixed-clock";
 			clock-frequency = <32000>;
@@ -159,6 +169,7 @@ booster: regulator-booster {
 	};
 
 	soc {
+		bootph-all;
 		compatible = "simple-bus";
 		#address-cells = <1>;
 		#size-cells = <1>;
@@ -197,6 +208,7 @@ ipcc: mailbox@4c001000 {
 		};
 
 		rcc: rcc@50000000 {
+			bootph-all;
 			compatible = "st,stm32mp1-rcc", "syscon";
 			reg = <0x50000000 0x1000>;
 			#clock-cells = <1>;
@@ -208,6 +220,7 @@ rcc: rcc@50000000 {
 		};
 
 		pwr_regulators: pwr@50001000 {
+			bootph-all;
 			compatible = "st,stm32mp1,pwr-reg";
 			reg = <0x50001000 0x10>;
 			st,tzcr = <&rcc 0x0 0x1>;
@@ -252,6 +265,7 @@ pwr_irq: pwr@50001020 {
 		};
 
 		pinctrl: pinctrl@50002000 {
+			bootph-all;
 			#address-cells = <1>;
 			#size-cells = <1>;
 			compatible = "st,stm32mp157-pinctrl";
@@ -260,6 +274,7 @@ pinctrl: pinctrl@50002000 {
 			hwlocks = <&hsem 0 1>;
 
 			gpioa: gpio@50002000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -271,6 +286,7 @@ gpioa: gpio@50002000 {
 			};
 
 			gpiob: gpio@50003000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -282,6 +298,7 @@ gpiob: gpio@50003000 {
 			};
 
 			gpioc: gpio@50004000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -293,6 +310,7 @@ gpioc: gpio@50004000 {
 			};
 
 			gpiod: gpio@50005000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -304,6 +322,7 @@ gpiod: gpio@50005000 {
 			};
 
 			gpioe: gpio@50006000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -315,6 +334,7 @@ gpioe: gpio@50006000 {
 			};
 
 			gpiof: gpio@50007000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -326,6 +346,7 @@ gpiof: gpio@50007000 {
 			};
 
 			gpiog: gpio@50008000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -337,6 +358,7 @@ gpiog: gpio@50008000 {
 			};
 
 			gpioh: gpio@50009000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -348,6 +370,7 @@ gpioh: gpio@50009000 {
 			};
 
 			gpioi: gpio@5000a000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -359,6 +382,7 @@ gpioi: gpio@5000a000 {
 			};
 
 			gpioj: gpio@5000b000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -370,6 +394,7 @@ gpioj: gpio@5000b000 {
 			};
 
 			gpiok: gpio@5000c000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -488,6 +513,7 @@ hdp: pinctrl@5002a000 {
 		};
 
 		pinctrl_z: pinctrl@54004000 {
+			bootph-all;
 			#address-cells = <1>;
 			#size-cells = <1>;
 			compatible = "st,stm32mp157-z-pinctrl";
@@ -496,6 +522,7 @@ pinctrl_z: pinctrl@54004000 {
 			hwlocks = <&hsem 0 1>;
 
 			gpioz: gpio@54004000 {
+				bootph-all;
 				gpio-controller;
 				#gpio-cells = <2>;
 				interrupt-controller;
@@ -576,6 +603,7 @@ usbh_ehci: usb@5800d000 {
 		};
 
 		ltdc: display-controller@5a001000 {
+			bootph-some-ram;
 			compatible = "st,stm32-ltdc";
 			reg = <0x5a001000 0x400>;
 			interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>,
@@ -587,6 +615,7 @@ ltdc: display-controller@5a001000 {
 		};
 
 		iwdg2: watchdog@5a002000 {
+			bootph-all;
 			compatible = "st,stm32mp1-iwdg";
 			reg = <0x5a002000 0x400>;
 			interrupts = <GIC_SPI 151 IRQ_TYPE_LEVEL_HIGH>;
@@ -645,6 +674,7 @@ rtc: rtc@5c004000 {
 		};
 
 		bsec: efuse@5c005000 {
+			bootph-all;
 			compatible = "st,stm32mp15-bsec";
 			reg = <0x5c005000 0x400>;
 			#address-cells = <1>;
@@ -667,6 +697,7 @@ ethernet_mac_address: mac@e4 {
 		};
 
 		etzpc: bus@5c007000 {
+			bootph-all;
 			compatible = "st,stm32-etzpc", "simple-bus";
 			reg = <0x5c007000 0x400>;
 			#address-cells = <1>;
-- 
2.53.0

